web/client/actions/widgets.js

```diff
@@ -248,6 +251,29 @@ const toggleDependencySelector = (active, settings) => setupDependencySelector({
     active,
     ...settings
 });
+/**
+ * Collapse/Expand the widget
+ * @param {object} widget the widget to collapse
+ */
+const toggleCollapse = (widget, target = DEFAULT_TARGET) => ({
+    type: TOGGLE_COLLAPSE,
+    widget,
+    target
+});
+
+/**
+ * Collapse/Expand all the widgets
+ */
+const toggleCollapseAll = (target = DEFAULT_TARGET) => ({
```

Comment:
test for this

Reply:
It's tested with the reducer, anyway, I need to add a test for the action itself